src/app/pages/login/login.component.ts

```diff
@@ -1,59 +1,70 @@
 import { Component } from '@angular/core';
-import { FormBuilder, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
-import { Router, RouterLink, RouterLinkActive } from '@angular/router';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router, RouterLink, RouterLinkActive, NavigationStart } from '@angular/router';
 import { AuthService } from '../../auth/auth.service';
-import { HttpClient, HttpHeaders, HttpErrorResponse } from '@angular/common/http';
-import { Observable, throwError, catchError, of } from 'rxjs';
-
-interface LoginResponse{
-  token?: string;
-}
+import { SharedModule } from '../../shared/shared.module';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Subscription } from 'rxjs';
+import { defaultSnackbarConfig } from '../../shared/configs/defaultSnackbarConfig';
 
 @Component({
   selector: 'app-login',
   standalone: true,
-  imports: [RouterLink, RouterLinkActive, FormsModule, ReactiveFormsModule],
+  imports: [RouterLink, RouterLinkActive, SharedModule],
   templateUrl: './login.component.html',
   styleUrl: './login.component.css'
 })
 export class LoginComponent {
   loginForm: FormGroup;
-  private apiUrl = 'https://rb-messenger.azurewebsites.net';
+  isInProgress: boolean = false;
 
-  constructor(private fb: FormBuilder, private authService: AuthService, private http: HttpClient, private router: Router) {
-    this.loginForm = this.fb.group({
+  private _routerSubscription!: Subscription;
+
+  constructor(private _fb: FormBuilder, private _authService: AuthService, private _router: Router,
+    private _snackBar: MatSnackBar
```

Comment:
To avoid duplicating of your code, you should create a SnackbarService, which will receive the message string and the type of message ('success' || 'warning'). Example:
![image](https://github.com/AndriyKuziv/rb-messenger-angular/assets/89141792/fa5e8959-062c-4d17-ab4b-c6d24f79e159)